phoenix-core/src/main/java/org/apache/phoenix/execute/MutationState.java

```diff
@@ -896,21 +906,31 @@ private void send(Iterator<TableRef> tableRefIterator) throws SQLException {
                     continue;
                 }
                 // Validate as we go if transactional since we can undo if a problem occurs (which is unlikely)
-                long serverTimestamp = serverTimeStamps == null ? validateAndGetServerTimestamp(tableRef,
-                        multiRowMutationState) : serverTimeStamps[i++];
+                long
+                        serverTimestamp =
+                        serverTimeStamps == null ?
+                                validateAndGetServerTimestamp(tableRef, multiRowMutationState) :
+                                serverTimeStamps[i++];
                 Long scn = connection.getSCN();
                 long mutationTimestamp = scn == null ? HConstants.LATEST_TIMESTAMP : scn;
                 final PTable table = tableRef.getTable();
-                Iterator<Pair<PName, List<Mutation>>> mutationsIterator = addRowMutations(tableRef,
-                        multiRowMutationState, mutationTimestamp, serverTimestamp, false, sendAll);
+                Iterator<Pair<PName, List<Mutation>>>
+                        mutationsIterator =
+                        addRowMutations(tableRef, multiRowMutationState, mutationTimestamp,
+                                serverTimestamp, false, sendAll);
                 // build map from physical table to mutation list
                 boolean isDataTable = true;
                 while (mutationsIterator.hasNext()) {
                     Pair<PName, List<Mutation>> pair = mutationsIterator.next();
                     PName hTableName = pair.getFirst();
                     List<Mutation> mutationList = pair.getSecond();
+                    if (clonedDataMutations.size() == 0) {
```

Comment:
clonedDataMutations are not used anymore. We can remove it.

Reply:
good catch